ultimarc/devices/ultrastik.py

```diff
@@ -0,0 +1,177 @@
+        data = (ct.c_ubyte * USTIK_PRE_MESG_LENGTH)(0)
+        data[0] = new_id
+
+        resp_1 = self.write_raw(USBRequestCode.ULTRASTIK_E9, 0x1, USTIK_PRE_INTERFACE, ct.c_void_p(), 0x0,
+                               request_type=USBRequestType.REQUEST_TYPE_VENDOR,
+                               recipient=USBRequestRecipient.RECIPIENT_OTHER)
+        # We need to write 32 bytes here.
+        resp_2 = self.write(USBRequestCode.ULTRASTIK_EB, 0x0, USTIK_PRE_INTERFACE, data, USTIK_PRE_MESG_LENGTH,
+                            request_type=USBRequestType.REQUEST_TYPE_VENDOR,
+                            recipient=USBRequestRecipient.RECIPIENT_OTHER)
+        resp_3 = self.read(USBRequestCode.ULTRASTIK_EA, 0x0, USTIK_PRE_INTERFACE, ct.c_void_p(), 0x0,
+                          request_type=USBRequestType.REQUEST_TYPE_VENDOR,
+                          recipient=USBRequestRecipient.RECIPIENT_OTHER)
+        resp_4 = self.write(USBRequestCode.ULTRASTIK_E9, 0x0, USTIK_PRE_INTERFACE, ct.c_void_p(), 0x0,
+                           request_type=USBRequestType.REQUEST_TYPE_VENDOR,
+                           recipient=USBRequestRecipient.RECIPIENT_OTHER)
+
+        return not False in [resp_1, resp_2, resp_3, resp_4]
+
+    def set_config(self, config_file: str) -> bool:
+        """ Set the joystick configuration """
+
+        config = JSONObject(self.validate_config_base(config_file, USTIK_RESOURCE_TYPES))
+
+        data = UltraStikStruct()
+        data.keepAnalog = 0x11 if config.keepAnalog else 0x11
+        data.mapSize = config.mapSize
+        data.restrictor = 0x09 if config.restrictor else 0x10
+        data.borders = (ct.c_uint8 * 8)(*config.borders)
+        data.map = (ct.c_uint8 * 81)(*[DIRECTION_MAP[v] for v in config.map])
+        data.flash = 0x00 if config.flash else 0xFF
+        data.reserved = (ct.c_uint8 * 3)(*[0, 0, 0])
+
+        resp_2 = resp_3 = resp_4 = False
+        payload = (ct.c_uint8 * USTIK_PRE_MESG_LENGTH)(0)
+
+        resp_1 = self.write_raw(USBRequestCode.ULTRASTIK_E9, 0x1, USTIK_PRE_INTERFACE, ct.c_void_p(), 0x0,
+                                request_type=USBRequestType.REQUEST_TYPE_VENDOR,
+                                recipient=USBRequestRecipient.RECIPIENT_OTHER)
+        # We need to write 32 bytes here.
+        for x in range(3):
+            ct.memmove(ct.addressof(payload), ct.byref(data, USTIK_PRE_MESG_LENGTH * x), USTIK_PRE_MESG_LENGTH)
+            _logger.debug(f"  config data block {x+1}: {' '.join('%02X' % b for b in payload)}")
+
+            resp_2 = self.write(USBRequestCode.ULTRASTIK_EB, 0x0, USTIK_PRE_INTERFACE,
+                                payload, USTIK_PRE_MESG_LENGTH,
+                                request_type=USBRequestType.REQUEST_TYPE_VENDOR,
+                                recipient=USBRequestRecipient.RECIPIENT_OTHER)
+            resp_3 = self.read(USBRequestCode.ULTRASTIK_EA, 0x0, USTIK_PRE_INTERFACE, ct.c_void_p(), 0x0,
```

Comment:
I think this needs to be write instead of read.

Reply:
I carefully reviewed the C code in the 'ultimarc-linux' for pre-2015 joysticks and this should be a read statement.  This code works correctly with my pre-2015 joystick.

Reply:
I'll review this...

Reply:
If these worked of you then I am good and we can resolve this comment